scripts/apps/dashboard/widget-react.tsx

```diff
@@ -37,9 +63,15 @@ export class WidgetReact extends React.PureComponent<IProps, IState> {
                         localStorage.removeItem('SIDE_WIDGET');
 
                         return initialState;
-                    } else {
-                        return undefined;
+                    } else if (closedIntentionally.value === true) {
+                        const prevWidgetState = widgetState[this.props.widget.active._id];
+
+                        if (prevWidgetState != null) {
```

Comment:
not all code paths return a value. add an `else` with `return undefined` just to be explicit